src/heuristicblocking.js

```diff
@@ -211,6 +211,13 @@ HeuristicBlocker.prototype = {
       snitch_map.setItem(tracker_origin, firstParties);
     }
 
+    var action_map = this.storage.getBadgerStorageObject('action_map');
+    // If tracker_origin is not yet in action_map, add it with action `constants.ALLOW`
+    if (!action_map.hasItem(tracker_origin)) {
```

Comment:
Ah, good point. In fact, `tracker_fqdn` needs to replace `tracker_origin` in a few of the calls.`snitch_map.hasItem()` should be called on the fqdn, as should the `snitch_map.setItem()` call. For this if clause, perhaps there should actually be two separate if clauses, one for `tracker_origin` and another for `tracker_fqdn`. Does that sound about right?